distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Internal/Existing_Enso_Asset.enso

```diff
@@ -71,15 +73,23 @@ type Existing_Enso_Asset
        Fetches the basic information about an existing file from the Cloud.
        It will fail if the file does not exist.
     get_asset_reference_for (file : Enso_File) -> Existing_Enso_Asset ! File_Error =
-        # TODO we will get this from the `resolve_path` endpoint, for now though we use manual resolution
-        current_user = Enso_User.current
-        if file.enso_path.organization_name != current_user.organization_name then Unimplemented.throw "Currently only resolving paths for the current user is supported." else
-            root = Existing_Enso_Asset.Value "" current_user.root_directory_id Enso_Asset_Type.Directory
-            file.enso_path.path_segments.fold root current_dir-> segment->
-                children = current_dir.list_directory . catch Illegal_Argument _->
-                    Error.throw (Illegal_Argument.Error current_dir.title+" is not a directory.")
-                (children.find c-> c.title == segment) . catch Not_Found _->
-                    Error.throw (File_Error.Not_Found file)
+        # TODO remove workaround for bug https://github.com/enso-org/cloud-v2/issues/1173
+        path = if file.enso_path.is_root then file.enso_path.to_text + "/" else file.enso_path.to_text
+        Existing_Enso_Asset.resolve_path path if_not_found=(Error.throw (File_Error.Not_Found file))
+
+    ## PRIVATE
+       Resolves a path to an existing asset in the cloud.
+    resolve_path (path : Text) ~if_not_found =
+        handle_not_found _ = if_not_found
+        error_handlers = Map.from_vector [["resource_missing", handle_not_found]]
+
+        uri = Utils.cloud_root_uri+"path/resolve"
+        payload = JS_Object.from_pairs [["path", path]]
+
+        # TODO remove workaround - this should be a Get endpoint, not Post
+        response = Context.Output.with_enabled <|
+            Utils.http_request_as_json HTTP_Method.Post uri payload error_handlers=error_handlers
+        Existing_Enso_Asset.from_json response
```

Comment:
This is the core change of this PR - instead of traversing the path and relying on `list_directory` on each part, potentially doing N big requests (returning the whole directory listing...), we now do just 1 request and the Cloud does all of the resolution for us.